R/endpoints.R

```diff
@@ -2111,21 +2115,31 @@ pvt_twitter <- function(
     auth,
     locations,
     ...,
-    dates = NULL,
-    epiweeks = NULL,
+    time_type = c("day", "week"),
+    time_values = "*",
     fetch_args = fetch_args_list()) {
   rlang::check_dots_empty()
 
+  time_type <- match.arg(time_type)
```

Comment:
I'm not entirely sure this is necessary, or else `pub_covidcast` is missing something important, since it doesn't do this. I would maybe make a function for this bit of logic if it is necessary, since it (should?) be included 3 times.

Reply:
Here it's necessary because the default value is `c("day", "week")`, but you can only have one `time_type`. `match.arg()` takes the first if the default is not overridden. It's a common convention in R that if the argument can be one of several string values, you make the default list all of them, and `match.arg()` takes care of picking the first (if none is provided) or ensuring the provided value is valid otherwise.

`pub_covidcast()` doesn't set a default for this argument, so it's not strictly necessary. However, `pub_covidcast()` could do `match.arg(time_type, c("day", "week"))` to enforce that the type must be one or the other but not both.